### ISSUE TITLE
Race condition between jobs started by eunomia

### ISSUE THREAD

**Issue by akavel** [KohlsTechnology/eunomia#179]
**What version of eunomia are you using?**

eunomia version: 0.0.4

**Does this issue reproduce with the latest release?**

yes

**What operating system and processor architecture are you using (`kubectl version`)?**

<details><summary><code>kubectl version</code> Output</summary><br><pre>
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.8", GitCommit:"211047e9a1922595eaa3a1127ed365e9299a6c23", GitTreeState:"clean", BuildDate:"2019-10-15T12:11:03Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:09:08Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
</pre></details>

**What did you do?**

As noticed by @safirh, there is race condition between jobs started by eunomia. Specifically, if two reconciliation requests for a CR (say, "A" and "B") are submitted after each other (esp. when the time between them is short; for example, as a result of two Webhook triggers from two commits), eunomia would *start* a new Job for the 2nd version of the CR (B) *before the Job for the 1st one (A) ended*. The two jobs would then be running concurrently, and with nondeterministic scheduling in Kubernetes, it can happen that the Job B will end (run resourceManager.sh) *before* Job A. As a result, Job A's "apply" would then override Job B's "apply", leaving cluster in old state.

This gets even more problematic when we implement a "status" field for the CRD in eunomia, as adding job info and git SHA1 info there would be confusing and misleading, with races between jobs running in parallel.

**What did you expect to see?**

I would expect that new job, for applying a newer version of a CR, would be postponed until the previous job (for an older version of the CR) finished (with success or failure).

**What did you see instead?**

Jobs for two versions of a CR from the same Git repository may run concurrently, leading to a race condition, where the result of applying the older version/commit of the CR may override the result of applying the newer version/commit of the CR.

**Comment by seanmalloy** [comment]
@akavel and @safirh thanks for the bug report! It looks like we need to fix this bug in the near future.

**Comment by seanmalloy** [comment]
@safirh the code changes are merged. I reopened the issue to see if we can write some unit or e2e tests before we officially close this issue.
